File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string text = get_string("Please input some text:\n");

    int words = 1;
    int letters = 0;
    int sentences = 0;

    //counting words
    for (int i = 0; i < strlen(text); i ++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters = letters + 1;
        }
        else if (text[i] == ' ')
        {
            words = words + 1;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences = sentences + 1;
        }

    }
    //algorithm

    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    //printing grade

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
def get_text():
    text = input("Text: ")
    return text

def coleman_algo(text):
    #count number of letter
    letter_count = 0
    word_count = 0
    for letter in (text):
        if letter.isalpha() == True:
            letter_count += 1
    for word in (text):
        if word == " ":
            word_count += 1
    word_count = word_count + 1
    #count number of letter per 100 words
    nolphw = (letter_count / word_count) * 100
    #count number of sentences
    number_of_sentences = 0
    for character in text:
        if character == "." or character == "?" or character == "!":
            number_of_sentences += 1
    #average number of sentence per 100 word
    anosph = (number_of_sentences / word_count) * 100

    #algorithm
    grade = int(round((0.0588 * nolphw) - (0.296 * anosph) - 15.8))
    return grade

def grade_indentification(grade):
    if grade >= 16:
        print("Grade 16+")
    elif grade <= 1:
        print("Before Grade 1")
    else:
        print(f'Grade {grade}')


in_text = get_text()
grade = coleman_algo(in_text)
grade_indentification(grade)
